src/js/options.js

```diff
@@ -224,6 +228,33 @@ function removeAllData() {
   }
 }
 
+function downloadCloud() {
+  chrome.runtime.sendMessage({type: "downloadCloud"},
+    function (status) {
+      if (status.success) {
+        alert(i18n.getMessage("download_cloud_success"));
+        reloadWhitelist();
+      } else {
+        alert(i18n.getMessage("download_cloud_failure"));
```

Comment:
The message handler in `webrequest.js` returns `{success: false, message: message}`, but the code here alerts with the "failure" message no matter what. This should check `status.message` and decide whether to alert with the failure message or the "no data" message.

Reply:
The message was mostly there for error debugging, but this is a reasonable enhancement.

Reply:
Ah, ok, makes sense. I do think it would be useful for the user to differentiate between the failure modes.

Reply:
After making this change, I'm not sure what would make the failure message (as opposed to the no data message) appear; write conditions can be manipulated to exceed quotas and produce errors, but read operations have no such quotas.  Presumably, loss of network connectivity during the transfer would do it, but I'm not sure how to reliably replicate that for such a small download ( <8kB by definition).  No network connectivity appears to fall back on local cache.